itou/www/apply/views/list_views.py

```diff
@@ -86,5 +103,70 @@ def list_for_siae(request, template_name="apply/list_for_siae.html"):
         job_applications, request.GET.get("page"), items_per_page=10
     )
 
-    context = {"siae": siae, "job_applications_page": job_applications_page}
+    context = {
+        "siae": siae,
+        "job_applications_page": job_applications_page,
+        "filters_form": filters_form,
+        "filters": filters,
+    }
     return render(request, template_name, context)
+
+
+##########################################################################
+######## Functions for internal-use only, not linked to a path. ##########
+##########################################################################
+
+
+def _add_filters_to_query(data, query):
```

Comment:
Désolé, j'arrive un peu tard là dessus…

La fonction `_add_filters_to_query` est fortement couplée aux données du formulaire `FilterJobApplicationsForm`, et peut difficilement être utilisée seule. Elle devrait faire partie des méthodes de nettoyage du formulaire : 

- https://docs.djangoproject.com/en/2.2/ref/forms/validation/#cleaning-a-specific-field-attribute
- https://docs.djangoproject.com/en/2.2/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other

Comme ça, tu sors tes données nettoyées de ton formulaire et tu construit ta requête directement dans la vue, tout est découplé.

On en parle si tu veux.

Reply:
Super ! Je viens de le changer. Tu me diras si c'est ok pour toi ?